remote/remote.go

```diff
@@ -193,14 +193,13 @@ func newV1Image(keychain authn.Keychain, repoName string, platform imgutil.Platf
 		time.Sleep(100 * time.Duration(i) * time.Millisecond) // wait if retrying
 		image, err = remote.Image(ref, remote.WithAuth(auth), remote.WithTransport(http.DefaultTransport), remote.WithPlatform(v1Platform))
 		if err != nil {
-			if err == io.EOF && i != maxRetries {
+			transportErr, ok := err.(*transport.Error)
```

Comment:
For some reason I find this code a bit difficult to read. What do you think of the following?
```
if err != nil {
			if err == io.EOF && i != maxRetries {
				continue // retry
			}
			if transportErr, ok := err.(*transport.Error); ok && is40x(transportErr) {
				if i != maxRetries {
					continue // retry
				}
				if len(transportErr.Errors) > 0 { // will this always be true for 40x ?
					return emptyImage(platform)
				}
			}
			if strings.Contains(err.Error(), "no child with platform") {
				return emptyImage(platform)
			}
			return nil, fmt.Errorf("connect to repo store '%s': %s", repoName, err.Error())
		}
```